File: samples/origami-simple-app/app/controllers/Application.java
package controllers;

import static play.data.Form.form;

import java.util.List;

import models.incidences.HasComment;
import models.vertices.Comment;
import models.vertices.Log;
import play.Logger;
import play.data.Form;
import play.mvc.Controller;
import play.mvc.Result;
import views.html.index;
import views.html.logs;
import views.txt.sql.getLogs;

import com.wingnest.play2.origami.GraphDB;
import com.wingnest.play2.origami.annotations.Transactional;
import com.wingnest.play2.origami.annotations.WithGraphDB;

@WithGraphDB
public class Application extends Controller {

	public static Result index() {
		return ok(index.render());
	}

	@Transactional
	public static Result postLog() {
		final Form<Log> logForm = form(Log.class).bindFromRequest();
		if ( logForm.hasErrors() ) {
			return badRequest(logForm.errorsAsJson());
		}
		logForm.get().save();
		return ok();
	}

	public static Result getLogs() {
		final String query = getLogs.render().body();
		final List<Log> logModels = GraphDB.synchQueryModel(query);
		Logger.debug(logModels.toString());
		return ok(logs.render(logModels));
	}

	@Transactional
	public static Result postComment(String logId) {
		final Form<Comment> commentForm = form(Comment.class).bindFromRequest();
		if ( commentForm.hasErrors() ) {
			return badRequest(commentForm.errorsAsJson());
		}
		if ( logId == null ) {
			return notFound();
		}

		boolean disupdateFlag = Boolean.parseBoolean(form().bindFromRequest().get("disupdateFlagLog"));

		Log log = GraphDB.findById(logId);
		if ( log == null )
			return notFound();
		log.disupdateFlag = disupdateFlag;

		HasComment hasComment = new HasComment();
		hasComment.setVertexes(log, commentForm.get());
		hasComment.save();

		return ok();
	}
}

File: samples/origami-simple-scala-app/app/controllers/Application.scala
package controllers

import play.api._
import play.api.mvc._
import play.api.data._
import play.api.data.Forms._
import models.vertices._
import models.incidences._

import com.wingnest.play2.origami._
import scala.collection.JavaConverters._

object Application extends Controller {

  def index = Action {
    Ok(views.html.index())
  }

  val logForm = Form(
    tuple(
      "title" -> nonEmptyText,
      "name" -> nonEmptyText,
      "logBody" -> nonEmptyText))

  def postLog = Action { implicit request =>
    logForm.bindFromRequest.fold(
      errors => BadRequest,
      {
        case (title, name, logBody) =>
          GraphDB.begin()
          try {
            val log = new Log()
            log.title = title
            log.name = name
            log.logBody = logBody
            log.save()
            GraphDB.commit()
          } catch {
            case e: Exception => {
              GraphDB.rollback()
            }
          } finally {
            GraphDB.close()
          }
          Ok
      })
  }

  def getLogs = Action {
    try {
      GraphDB.open()
      val query = views.txt.sql.getLogs.render().body
      val logModels: java.util.List[Log] = GraphDB.synchQueryModel(query)
      Ok(views.html.logs.render(logModels))
    } finally {
      GraphDB.close()
    }
  }

  val commentForm = Form(
    tuple(
      "name" -> nonEmptyText,
      "content" -> nonEmptyText,
      "disupdateFlagLog" -> boolean))

  def postComment(logId: String) = Action { implicit request =>
    commentForm.bindFromRequest.fold(
      errors => BadRequest,
      {
        case (name, content, disupdateFlagLog) =>
          GraphDB.begin()
          try {
            val log: Log = GraphDB.findById(logId);
            if (log == null)
              NotFound
            log.disupdateFlag = disupdateFlagLog;

            val comment = new Comment()
            comment.name = name
            comment.content = content

            val hasComment = new HasComment()
            hasComment.setVertexes(log, comment)
            hasComment.save()

            GraphDB.commit()
          } catch {
            case e: Exception => {
              GraphDB.rollback()
            }
          } finally {
            GraphDB.close()
          }
          Ok
      })
  }

}